.github/workflows/build-for-release.yml

```diff
@@ -9,37 +9,50 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macos-latest]
+        build_name:
+          [linux-x64-musl, linux-arm64-musl, macos-arm64, windows-x64-msvc]
         include:
-          - os: ubuntu-latest
+          - build_name: linux-x64-musl
+            os: ubuntu-latest
             exe_suffix: ""
             name_suffix: ""
             target: x86_64-unknown-linux-musl
-          - os: macos-latest
+          - build_name: linux-arm64-musl
+            os: ubuntu-latest
+            exe_suffix: ""
+            name_suffix: ""
+            target: aarch64-unknown-linux-musl
+          - build_name: macos-arm64
+            os: macos-latest
             exe_suffix: ""
             name_suffix: ".arm"
             target: aarch64-apple-darwin
-          - os: windows-latest
+          - build_name: windows-x64-msvc
+            os: windows-latest
             exe_suffix: ".exe"
             name_suffix: ".exe"
             target: x86_64-pc-windows-msvc
     steps:
       - uses: actions/checkout@v3
-      - name: Update rust
-        run: rustup update
-      - name: Add target
-        run: rustup target add ${{ matrix.target }}
-      - name: Cache dependencies
-        uses: Swatinem/rust-cache@v2
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: ${{ matrix.target }}
+          override: true
       - name: Build
-        run: "cargo build --release --locked --verbose --target ${{ matrix.target }}"
+        uses: actions-rs/cargo@v1
+        with:
+          use-cross: ${{ matrix.os != 'windows-latest' }}
```

Comment:
What's this for?

Reply:
Cross-compiling  on `ubuntu-latest` (x86_64) fails due to:

```
  = note: /usr/bin/ld: /home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crt1.o: Relocations in generic ELF (EM: 183)
          /usr/bin/ld: /home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crt1.o: Relocations in generic ELF (EM: 183)
          /usr/bin/ld: /home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crt1.o: Relocations in generic ELF (EM: 183)
          /usr/bin/ld: /home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crt1.o: Relocations in generic ELF (EM: 183)
          /usr/bin/ld: /home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crt1.o: Relocations in generic ELF (EM: 183)
          /usr/bin/ld: /home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crt1.o: Relocations in generic ELF (EM: 183)
          /usr/bin/ld: /home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crt1.o: Relocations in generic ELF (EM: 183)
          /usr/bin/ld: /home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crt1.o: error adding symbols: file in wrong format
          collect2: error: ld returned 1 exit status
          
``` 

Full logs: https://github.com/orhun/dotter/actions/runs/7142628993/job/19452312103

So we use [cross](https://github.com/cross-rs/cross). It is not needed for Windows for it is disabled.

Reply:
I see, since the old builds work without it maybe its better to change the value to `matrix.build_name == 'linux-arm64-musl` ?

And instead of `actions-rs/cargo` we can just conditionally run `cargo install cross` before the `cargo build` - I've checked the code and this is essentially what it does.

Reply:
Surely one day github will add arm runners, in the meantime we need to do stuff like this 🫠

Reply:
Done in ebdc7a1c37c10aa51f332b7c5f11ddaa1e36bde6

Reply:
https://github.com/orhun/dotter/actions/runs/7158490996/job/19490635099

If you expand the `run _cargo` part, looks like it isn't interpolating the value from the matrix properly `if [ "" = "linux-arm64-musl" ]`

IDK what's the right way to do this, looking

Reply:
That error is fixed in this branch now. It was due to using `build.name` instead of `build_name`.

Reply:
IDK, this looks correct -_-

Maybe the quotes are messing with it? I'd try with `matrix.target` since it's working (on the last line)

Reply:
I mean it is already fixed - I just didn't create new release. I amended the commit and pushed to this branch directly instead.

Created another release for verifying the fix: https://github.com/orhun/dotter/releases/tag/v0.13.0-test10
Workflow run: https://github.com/orhun/dotter/actions/runs/7158564700/job/19490807722